src/pages/login-page/LoginPage.tsx

```diff
@@ -1,39 +1,108 @@
-import { useEffect } from 'react';
-import { useAppDispatch } from '@metis/store/hooks';
+import SquareGroup from '@metis/assets/images/misc/square-group.png';
+import Modal from '@metis/common/components/ui/Modal';
+import constants from '@metis/common/configuration/constants';
+import connectSocket from '@metis/common/services/socket.service';
+import useMetamask from '@metis/features/auth/hooks/useMetamask';
+import MetaMaskService from '@metis/features/auth/services/metamask.service';
 import { login } from '@metis/features/auth/store/auth.actions';
-import Button from '@mui/material/Button';
+import {
+  setIsCreatingAccount,
+  setJupAccount,
+  setLoggedIn,
+} from '@metis/features/auth/store/auth.slice';
+import { useAppDispatch, useAppSelector } from '@metis/store/hooks';
+import { openToast } from '@metis/store/ui/ui.slice';
+import PeopleIcon from '@mui/icons-material/People';
+import { LoadingButton } from '@mui/lab';
 import Box from '@mui/material/Box';
 import Container from '@mui/material/Container';
-import SquareGroup from '@metis/assets/images/misc/square-group.png';
-import useMetamask from '@metis/features/auth/hooks/useMetamask';
+import { useEffect, useState } from 'react';
 import useStyles from './LoginPage.styles';
 
+type SignUpSuccessfulEventResponse = {
+  createdAt: number;
+  token: string;
+  address: string;
+  alias: string;
+};
+
 const LoginPage = () => {
   const dispatch = useAppDispatch();
   const classes = useStyles();
   const { account, connect } = useMetamask();
+  const { isConnectingToMetamask, isCreatingAccount } = useAppSelector((state) => state.auth);
 
   const handleLogin = async () => {
     await connect();
   };
 
   useEffect(() => {
     if (account) {
+      const socket = connectSocket({
```

Comment:
we could explore the option of creating a socket hook here, so we can reuse most of the logic